java/src/main/java/ai/rapids/cudf/Cuda.java

```diff
@@ -39,9 +45,9 @@
     protected boolean cleanImpl(boolean logErrorIfNotClean) {
       boolean neededCleanup = false;
       long origAddress = stream;
-      if (stream != 0) {
+      if (stream != defaultStream()) {
```

Comment:
This should check for any stream that shouldn't be cleaned up which includes implicit default stream (0), legacy default stream (1), and per-thread default stream (2).

Reply:
Done.